lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Sign.hs

```diff
@@ -226,16 +234,45 @@ estimateKeyWitnessCount utxo tx =
         in
             nonInputWits <> inputWits
   where
-    CardanoApi.Tx txbody@(CardanoApi.TxBody txbodycontent) _keyWits
+    CardanoApi.Tx (CardanoApi.TxBody txbodycontent) _keyWits
         = toCardanoApiTx tx
 
-    scripts = case txbody of
-        CardanoApi.ShelleyTxBody _ _ shelleyBodyScripts _ _ _ ->
-            shelleyBodyScripts
-        CardanoApi.ByronTxBody {} ->
-            error "estimateKeyWitnessCount: ByronTxBody"
+    timelockScriptVkCount :: Natural
+    timelockScriptVkCount = sum $ Map.elems $ Map.unionWith
+        (\_est spec -> spec)
+        upperBoundEstimatedTimelockVkCounts
+        specifiedTimelockVkCounts
```

Comment:
Since ordinary `Map.union` is left-biased, we could simplify this:

```suggestion
    timelockScriptVkCount = sum $ Map.elems $ Map.union
        specifiedTimelockVkCounts
        upperBoundEstimatedTimelockVkCounts
```

Reply:
I thought I'd be overly explicit here, as the behavior in case of conflicts is important.